IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/StateContext.cs

```diff
@@ -16,10 +17,23 @@ public class StateContext : MonoBehaviour
         /// </summary>
         public StateContext()
         {
+            this.Buttons = new ButtonHandler();
             this.CurrentState = new NeutralState(this);
+            this.SwapObject(Resources.Load<GameObject>("Objects/DefaultObject/Instance"));
+            this.SelectedBuilding.SetActive(false);
         }
 
         /// <summary>
+        /// Gets or sets the selected building
+        /// </summary>
+        public GameObject SelectedBuilding { get; set; }
```

Comment:
Why does the statecontext store information about a selected building. If multiple or all states need this property it should be shared via the AbstractState or a child of the AbstractState